solutions/fully-configurable/main.tf

```diff
@@ -1,7 +1,40 @@
 locals {
-  prefix = var.prefix != null ? (trimspace(var.prefix) != "" ? "${trimspace(var.prefix)}-" : "") : ""
+  prefix    = var.prefix != null ? (trimspace(var.prefix) != "" ? "${trimspace(var.prefix)}-" : "") : ""
+  crn_parts = var.existing_kms_key_crn != null ? split(":", var.existing_kms_key_crn) : null
+
+  kms_instance_crn = var.existing_kms_instance_crn != null ? var.existing_kms_instance_crn : var.existing_kms_key_crn != null ? "${join(":", slice(local.crn_parts, 0, length(local.crn_parts) - 2))}::" : null
+  kms_key_crn      = var.enable_kms_encryption ? (var.existing_kms_key_crn != null ? var.existing_kms_key_crn : module.kms[0].keys[format("%s.%s", local.kms_key_ring_name, local.kms_key_name)].crn) : null
+
+  kms_region        = var.enable_kms_encryption && (var.existing_kms_instance_crn != null || var.existing_kms_key_crn != null) ? module.existing_kms_crn_parser[0].region : null
+  kms_service_name  = var.enable_kms_encryption && (var.existing_kms_instance_crn != null || var.existing_kms_key_crn != null) ? module.existing_kms_crn_parser[0].service_name : null
+  kms_instance_guid = var.enable_kms_encryption && (var.existing_kms_instance_crn != null || var.existing_kms_key_crn != null) ? module.existing_kms_crn_parser[0].service_instance : null
+  kms_account_id    = var.enable_kms_encryption && (var.existing_kms_instance_crn != null || var.existing_kms_key_crn != null) ? module.existing_kms_crn_parser[0].account_id : null
+
+  kms_key_ring_name = var.app_config_key_ring_name != null ? "${local.prefix}${var.app_config_key_ring_name}" : null
+  kms_key_name      = var.app_config_key_name != null ? "${local.prefix}${var.app_config_key_name}" : null
+
```

Comment:
why are we not using `"terraform-ibm-modules/common-utilities/ibm//modules/crn-parser"` for CRN parsing here?

Reply:
I didnt mean this line exactly, I meant in general. But I see we are actually using it, so ignore